src/account-info/CustomProfileFields.js

```diff
@@ -0,0 +1,132 @@
+// @flow strict-local
+import * as React from 'react';
+import { View } from 'react-native';
+
+import { type UserOrBot, type UserId } from '../api/modelTypes';
+import WebLink from '../common/WebLink';
+import ZulipText from '../common/ZulipText';
+import ZulipTextIntl from '../common/ZulipTextIntl';
+import { ensureUnreachable } from '../generics';
+import { useSelector } from '../react-redux';
+import { tryGetUserForId } from '../selectors';
+import {
+  type CustomProfileFieldValue,
+  getCustomProfileFieldsForUser,
+} from '../users/userSelectors';
+import UserAvatar from '../common/UserAvatar';
+
+type Props = {|
+  +user: UserOrBot,
+|};
+
+function CustomProfileFieldUser(props: {| +userId: UserId |}): React.Node {
+  const { userId } = props;
+  const user = useSelector(state => tryGetUserForId(state, userId));
+
+  const styles = React.useMemo(
+    () => ({
+      text: { marginStart: 8 },
+      row: { flexDirection: 'row', marginVertical: 4 },
+    }),
+    [],
+  );
+
+  if (!user) {
+    return <ZulipTextIntl text="(unknown user)" />;
+  }
+
+  return (
+    <View style={styles.row}>
+      <UserAvatar size={24} avatarUrl={user.avatar_url} />
+      <ZulipText style={styles.text} text={user.full_name} />
+    </View>
+  );
```

Comment:
Here's an iOS screenshot:

<img width=400 src="https://user-images.githubusercontent.com/22248748/172237696-a1b735ab-c079-42c1-b754-d233de602166.png" />

I noticed a few things about the user item in the "users" field type:

- Could we vertically align the center of the avatar with the center of the user's name? The text looks higher than it should be.
- Should it be a link to the user's profile?
- Should we use a placeholder image and text if you've muted the user, like we do elsewhere?
- (Show email? Probably not; takes too much space to be worth it.)
- Should we show the user's status emoji, if set?

Ah and one more:

<img width=400 src="https://user-images.githubusercontent.com/22248748/172237313-ad62f98d-3787-4e06-a4d6-dc43210efac9.png" />

- This is obviously a contrived example where the name is very long and without spaces. Should we handle long names by truncating or wrapping (and if wrapping, where should we wrap)?

I'm wary of making the wrong abstraction, but I wonder if this is a case where something like `UserItem` should take care of handling all these subtle things, and grow a few more controls to handle callers' needs (probably one that lets us make the avatar smaller than 48, for a start). Just a thought.

Reply:
> I noticed a few things about the user item in the "users" field type:

Yeah, I agree with all of those. Good catch re: muted user.



> I wonder if this is a case where something like `UserItem` should take care of handling all these subtle things, and grow a few more controls to handle callers' needs (probably one that lets us make the avatar smaller than 48, for a start).

Yeah, I was thinking the same thing midway through reading your comment. :slightly_smiling_face: Will take a look.